### DISCUSSION TITLE
Form action changes protocol from http to https

### DISCUSSION THREAD

**Discussion by Stadly** [nextauthjs/next-auth#10345] [Help]
I'm using Auth.js with SvelteKit. In my development environment, everything works as it should. I have a `Dockerfile` for setting up the production environment. When the production environment is deployed it works as it should. But when testing the production environment locally, it does not work.

When I run the production environment locally, it is available under `http://localhost:3000`. The issue is that when I go to `http://localhost:3000/auth/signin`, the form looks like this: `<form action="https://localhost:3000/auth/signin/azure-ad" method="POST">`. As you can see, `http` has been changed to `https`. This results in a `ERR_SSL_PROTOCOL_ERROR`. How can I prevent this from happening?

When running the development environment without using the `Dockerfile` (`npm run dev`), the form action retains the `http` protocol, so it works as it should.

**Comment by c1969** [comment]
Hey  @Stadly, 
same Prob here... did get any results/solution on that meanewhile?

**Reply by Stadly** [reply]
No, have not been able to solve it.

**Comment by vortura** [comment]
I was stuck on this too. What worked for me was setting the `ORIGIN` environment variable to `http://localhost:3000`. E.g.
```
ORIGIN=http://localhost:3000 node build
```

See https://kit.svelte.dev/docs/adapter-node#environment-variables-origin-protocolheader-hostheader-and-port-header for details.
